wallet/tests/wallet.rs

```diff
@@ -470,18 +490,28 @@ macro_rules! assert_fee_rate {
                 }
         )*
 
-            #[allow(unused_mut)]
+        #[allow(unused_mut)]
         #[allow(unused_assignments)]
         let mut dust_change = false;
         $(
             $( $dust_change )*
                 dust_change = true;
         )*
 
-            let fee_amount = psbt
+        let fee_amount = psbt
```

Comment:
Can we use `Psbt::fee` for this?

```suggestion
        let fee_amount = psbt.fee().expect("failed to calculate fee");
```